subprojects/gst-plugins-base/gst-libs/gst/gl/web/gstglwindow_canvas.c

```diff
@@ -0,0 +1,172 @@
+/*
+ * GStreamer
+ * Copyright (C) 2024 Jorge Zapata <jzapata@fluendo.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin St, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <emscripten/html5.h>
+#include <gst/gst.h>
+#include <gst/gl/gl.h>
+#include "gstglcontext_emscripten.h"
+#include "gstglwindow_canvas.h"
+#include "../gstglwindow_private.h"
+
+#define GST_CAT_DEFAULT gst_gl_window_debug
```

Comment:
Why not gst_gl_window_canvas_debug?

Reply:
This is part of gstlgwindow_private. Is how other window systems work. They reuse the generic window category.